File: server/app/models/field.model.ts
import { Schema, Types, model } from 'mongoose';

const FieldSchema: Schema = new Schema(
    {
        userID: { type: Types.ObjectId, ref: 'User', required: true },
        name: { type: String, max: 30, trim: true, required: true },
        views: { type: Number, default: 0 },
        price: { type: Number, required: true, default: 0 },
        coverImage: { type: String, default: '' },
        type: {
            type: Number,
            required: true,
            default: 5,
            max: 11,
        },
        isPublic: { type: Boolean, default: true, required: true }, // true is public
        isLock: { type: Boolean, default: true, required: true }, // true is public
        isRepair: { type: Boolean, default: false, required: true }, // true is repair
        description: { type: String, default: '', max: 3000 },
        length: { type: Number, required: true },
        width: { type: Number, required: true },
        lock: {
            status: { type: Boolean, default: false },
            content: { type: String, default: '', max: 1500 },
        },
    },
    {
        timestamps: true,
        versionKey: false,
    }
);

export default model('Field', FieldSchema);

File: widget_component/lib/models/field_model.dart
class FieldModel {
  Lock? lock;
  String? sId;
  String? userID;
  String? name;
  int? views;
  int? price;
  String? coverImage;
  int? type;
  bool? isPublic;
  bool? isLock;
  bool? isRepair;
  String? description;
  int? length;
  int? width;
  String? createdAt;
  String? updatedAt;

  FieldModel(
      {this.lock,
      this.sId,
      this.userID,
      this.name,
      this.views,
      this.price,
      this.coverImage,
      this.type,
      this.isPublic,
      this.isLock,
      this.isRepair,
      this.description,
      this.length,
      this.width,
      this.createdAt,
      this.updatedAt});

  FieldModel.fromJson(Map<String, dynamic> json) {
    lock = json['lock'] != null ? Lock.fromJson(json['lock']) : null;
    sId = json['_id'];
    userID = json['userID'];
    name = json['name'];
    views = json['views'];
    price = json['price'];
    coverImage = json['coverImage'];
    type = json['type'];
    isPublic = json['isPublic'];
    isLock = json['isLock'];
    isRepair = json['isRepair'];
    description = json['description'];
    length = json['length'];
    width = json['width'];
    createdAt = json['createdAt'];
    updatedAt = json['updatedAt'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    if (lock != null) {
      data['lock'] = lock!.toJson();
    }
    data['_id'] = sId;
    data['userID'] = userID;
    data['name'] = name;
    data['views'] = views;
    data['price'] = price;
    data['coverImage'] = coverImage;
    data['type'] = type;
    data['isPublic'] = isPublic;
    data['isLock'] = isLock;
    data['isRepair'] = isRepair;
    data['description'] = description;
    data['length'] = length;
    data['width'] = width;
    data['createdAt'] = createdAt;
    data['updatedAt'] = updatedAt;
    return data;
  }
}

class Lock {
  bool? status;
  String? content;

  Lock({this.status, this.content});

  Lock.fromJson(Map<String, dynamic> json) {
    status = json['status'];
    content = json['content'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['status'] = status;
    data['content'] = content;
    return data;
  }
}
